src/equations/polytropic_euler_2d.jl

```diff
@@ -0,0 +1,172 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+    @doc raw"""
+        PolytropicEulerEquations2D(gamma, kappa)
+
+    The polytropic Euler equations
+    ```math
+    \partial t
+    \begin{pmatrix}
+    \rho \\ \rho v_1 \\ \rho v_2
+    \end{pmatrix}
+    +
+    \partial x
+    \begin{pmatrix}
+     \rho v_1 \\ \rho v_1^2 + \kappa\rho^\gamma \\ \rho v_1 v_2
+    \end{pmatrix}
+    +
+    \partial y
+    \begin{pmatrix}
+    \rho v_2 \\ \rho v_1 v_2 \\ \rho v_2^2 + \kappa\rho^\gamma
+    \end{pmatrix}
+    =
+    \begin{pmatrix}
+    0 \\ 0 \\ 0
+    \end{pmatrix}
+    ```
+    for an ideal gas with ratio of specific heats `gamma`
+    in two space dimensions.
+    Here, ``\rho`` is the density and ``v_1`` and`v_2` the velocities and
+    ```math
+    p = \kappa\rho^\gamma
+    ```
+    the pressure, which we replaced using this relation.
+
+    """
+    struct PolytropicEulerEquations2D{RealT <: Real, RealT <: Real} <:
+           AbstractPolytropicEulerEquations{2, 3}
+        gamma::RealT               # ratio of specific heats
+        kappa::RealT               # fluid scaling factor
+        inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+
+        function PolytropicEulerEquations2D(gamma, kappa)
+            new{typeof(gamma), typeof(kappa)}(gamma, kappa)
```

Comment:
```suggestion
        function PolytropicEulerEquations2D(gamma, kappa)
            gamma_, kappa_ = promote(gamma, kappa)
            new{typeof(gamma), typeof(kappa)}(gamma_, kappa_)
```
Such that compatible but different types can be used together